static/src/javascripts/projects/journalism/modules/submit-form.js

```diff
@@ -0,0 +1,57 @@
+// @flow
+import fastdom from 'lib/fastdom-promise';
+import fetch from 'lib/fetch';
+
+const isCheckbox = element => element.type === 'checkbox';
+const isValid = element => element.name !== '';
+
+const formToObj = elements =>
+    [].reduce.call(
+        elements,
+        (data, element) => {
+            if (isValid(element)) {
+                if (isCheckbox(element)) {
+                    data[element.name] = (data[element.name] || []).concat(
+                        element.value
+                    );
+                } else {
+                    data[element.name] = element.value;
+                }
+            }
+            return data;
+        },
+        {}
+    );
+
+const showConfirmation = cForm => {
+    fastdom.write(() => {
+        cForm.textContent = 'Thank you for your contribution';
+    });
+};
+
+const showError = cForm => {
+    fastdom.write(() => {
+        cForm.append('Sorry there was an error submitting your contribution');
```

Comment:
Perhaps we can add a bit of punctuation here, e.g. a comma after "Sorry". Additionally, it would be nice to let the user try again.

Reply:
Yes, however I unfortunately don't know what would have gone wrong so wouldn't be able to give them a meaningful error, I'll come back to this